drivers/sensor/bh1750/bh1750.c

```diff
@@ -0,0 +1,145 @@
+			       struct sensor_value *val)
+{
+	struct bh1750_data *data = dev->data;
+	
+	if (chan != SENSOR_CHAN_LIGHT) {
+		return -ENOTSUP;
+	}
+	
+	val->val1 = (int32_t)(data->ambient_light);
+	val->val2 = (data->ambient_light - val->val1) * 1000000;
+
+	return 0;
+}
+
+
+static int bh1750_sample_fetch(const struct device *dev,
+				enum sensor_channel chan)
+{
+	struct bh1750_data *drv_data = dev->data;
+	const struct bh1750_config *cfg = dev->config;
+	uint8_t buf[2];
+	int rc = 0;
+
+	// In case of one time measurement we need to start a new measurement
+#ifdef CONFIG_BH1750_MODE_ONE_TIME
+	uint8_t buf_ctrl = BH1750_CTRL_MEAS;	
+	rc = i2c_write_dt(&cfg->i2c, &buf_ctrl, 1);
+	if(rc < 0){
+		LOG_ERR("Error writing in i2c bus: %d", rc);
+		return -ENXIO;
+	} else {
+		LOG_DBG("Starting a new measurement: %d", buf_ctrl);
+	}
+
+	// We need to wait for the new measeurement to be taken
+	#ifdef CONFIG_BH1750_H_RESOLUTION_MODE
+		k_msleep(120);
+	#elif CONFIG_BH1750_H_RESOLUTION_MODE2
+		k_msleep(120);
+	#elif CONFIG_BH1750_L_RESOLUTION_MODE
+		k_msleep(16);
+	#endif
+#endif
+
+	// read raw value data
+	rc = i2c_read_dt(&cfg->i2c, buf,2);
+
+	if (rc < 0) {
+		LOG_ERR("Failed to read data sample.");
+		return -EIO;
+	}
+
+	// We need to divide by 1.2 since it is the accuracy according to datasheet
+	float data_value = (((uint16_t)buf[0] << 8) | buf[1]) / 1.2;
+
+	// We need to divide by 2 in case of resolution mode 2
+	#ifdef CONFIG_BH1750_H_RESOLUTION_MODE2
+		data_value /= 2;
+	#endif
+
+	LOG_DBG("Light value: %f", data_value);
+	drv_data->ambient_light = data_value;
+
+	return 0;
+}
+
+static const struct sensor_driver_api bh1750_driver_api = {
+	.sample_fetch = bh1750_sample_fetch,
+	.channel_get = bh1750_channel_get,
+};
+
+
+
+int bh1750_init(const struct device *dev)
+{
+	const struct bh1750_config *cfg = dev->config;
+	int rc = 0;
+
+	if (!device_is_ready(cfg->i2c.bus)) {
+		LOG_ERR("Bus device is not ready");
+		return -ENODEV;
+	} else {
+		LOG_INF("Bus device is ready");
+	}
+
+	// Set mode and resolution
+	uint8_t buf = BH1750_CTRL_MEAS;	
+
+	rc = i2c_write_dt(&cfg->i2c, &buf, 1);
+	if(rc < 0){
+		LOG_ERR("Error writing in i2c bus: %d", rc);
+		return -ENXIO;
+	} else {
+		LOG_INF("The mode was set correctly: 0x%x", BH1750_CTRL_MEAS);
+	}
+
+	return 0;
+}
+
+
+#define BH1750_DEFINE(inst)									                        \
+	static struct bh1750_data bh1750_data_##inst;					                \
+												                                    \
+	/* pull instance configuration from Devicetree */                               \
+	static const struct bh1750_config bh1750_config_##inst = {			            \
+		.i2c = I2C_DT_SPEC_INST_GET(inst)						                    \
+	};						    				                                    \
+												                                    \
+    /* define a new device inst. with the data and config defined above*/           \
+	DEVICE_DT_INST_DEFINE(inst, bh1750_init, NULL,						            \
```

Comment:
SENSOR_DT_INST_DEFINE

Reply:
I think you mean `SENSOR_DEVICE_DT_INST_DEFINE`. 

I propose the following changes:
```suggestion
	SENSOR_DEVICE_DT_INST_DEFINE(inst, bh1750_init, NULL,			      \
```